nbconvert/templates/markdown.tpl

```diff
@@ -8,7 +8,14 @@
 {%- endblock output_prompt %}
 
 {% block input %}
-```{% if nb.metadata.language_info %}{{ nb.metadata.language_info.name }}{% endif %}
+```
+{%- if 'magics_language' in cell.metadata  -%}
+    {{ cell.metadata.magics_language}}
+{%- elif 'pygments_lexer' in nb.metadata.get('language_info', {}) -%}
+    {{ nb.metadata.language_info.pygments_lexer }}
```

Comment:
I don't think markdown is usually rendered by something using pygments for highlighting, so we shouldn't be giving priority to the pygments lexer name here.